pkg/controller/control.go

```diff
@@ -1156,6 +1167,77 @@ func (c *Controller) GetLatestMetics() *ptypes.Metrics {
 	return metrics
 }
 
+func (c *Controller) ShowProfiler() (string, bool, error) {
+	c.profilerLock.Lock()
+	defer c.profilerLock.Unlock()
+
+	logrus.Infof("Prepare to do ShowProfilerConfig")
+	if c.profiler.ProfilerServer != nil {
+		return c.profiler.ProfilerServer.Addr, true, nil
+	}
+	return "", true, nil
+}
+
+func (c *Controller) EnableProfiler() (string, bool, error) {
+	c.profilerLock.Lock()
+	defer c.profilerLock.Unlock()
+
+	logrus.Infof("Prepare to do EnableProfiler")
+	var err error
+	if c.profiler.ProfilerEnabled {
+		return c.profiler.ProfilerServer.Addr, true, nil
+	}
+	// grpcProt is unique, we could use it as the base port of profiler
+	grpcPort, err := strconv.Atoi(strings.Split(c.GRPCAddress, ":")[1])
+	if err != nil {
+		logrus.Warnf("Failed to convert the gRPC port to integer.")
+		return "", false, err
+	}
+	profilerPort := profilerPortBase + grpcPort
```

Comment:
hi @shuo-wu, thanks for the review.
Yes, the profiler port is the sum of the grpcPort and the base (6060).

We have the unique gPRC server on each im-e pod, so unless we have other processes running on the im-e pod and use this port. It should not conflict.  Or I could try to add --profilerPort to the profiler enable command. To give users a chance to define which port they wanted. WDYT?

If the port is already binding by others, ListenAndServe will return an error. We may need to check the log to know that.

Reply:
In v1 volume implementation, **each** engine or replica is running as a separate process and **using some ports** inside the IM pod. And by design, [port 10000~30000 is reserved for engines and replicas](https://github.com/longhorn/longhorn-instance-manager/blob/f80354ec9a09e986f378017410891b26dddecc00/app/cmd/start.go#L49-L56) in IM pods. Hence, an engine or replica processing may be using port 16060. 
I am thinking if we can use more smaller number (<10000) as the 1st profiler port instead. Or maybe you can directly use 6060.

Reply:
Hi @shuo-wu, Thanks for the explanation. I do not find this reservation.

OK, that I could use the profiler port base `20001`. Then, the profiler server will listen on the engine/replica port + 20001.
That means we will use 30001 ~ 50001 for the profiler server.
WDYT?

Reply:
I am fine with using either start port 30001+ or 6060+.
BTW, there is one minor issue: Each replica process would take 10 port numbers. Then for replica IM pods, the replica profiler ports will be like: 30001, 30011, 30021...... It's a little bit of waste

Reply:
Hi @shuo-wu 
Because I would like to let users know that the gRPC port + 20001 would be the profiler server port (if enabled), I would like to keep that mechanism. 
Yes, that would be weird about the profiler port for the replica side. (Means 30001, 30011, 30021). But if the user would like to define the profiler server port continuously, they could use `--port` argument.
What do you think of that? Thanks!

Reply:
If you prefer not to use the port allocator in the IM server, I am fine with the current implementation.

Reply:
OK, thanks for it!
Because the profiler server is not enabled by default, I would like to make the mechanism of the profiler server port on the internal engine instead of by IM server.

In general cases, we might enable the profiler server for a while (for profiling) and then disable it.

Feel free to let me know if there is something I do not consider. I would like to refine this!